### PR TITLE
[DISCO-2310] Output a JSON file for contile to consume

### PR DESCRIPTION
**Pull request by taddes**
## References

JIRA: [DISCO-2310](https://mozilla-hub.atlassian.net/browse/DISCO-2310)
GitHub: [#104](https://github.com/mozilla-services/consvc-shepherd/issues/104)

## Description
We want Shepherd to produce a JSON file that gets sent to the Contile bucket.

Note there’s a dependency on [this PR](https://github.com/mozilla-services/consvc-shepherd/pull/96/files)

This requires:
- Validate allocation schema in `test_schema.py`
- Adding to_dict method in the AllocationSetting Model
- Adding to_dict method in the PartnerAllocation Model
-  An admin action similar to [the publish snapshot action](https://github.com/mozilla-services/consvc-shepherd/blob/main/consvc_shepherd/admin.py#L12)* to publish new allocation settings.
- Admin validation and construction of object to then be output as JSON
-  Added tests to test our new JSON schema for allocation

[DISCO-2310]: https://mozilla-hub.atlassian.net/browse/DISCO-2310?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: schema/allocation.schema.json [hunk 1]
```diff
@@ -8,49 +8,53 @@
...
-                        }
+            "minItems": 1,
+            "items": {
+                "type": "object",
+                "properties": {
+                    "position": {
+                        "type": "integer",
+                        "description": "1-based position of tile from left side, 1 for the first slot.",
+                        "minimum": 1,
+                        "maximum": 6
```
Comment (tiftran) [inline schema/allocation.schema.json:26]: i think we also wanted @ncloudioj to chime in on whether we wanted to restrict the number of allocations

Comment (ncloudioj) [reply schema/allocation.schema.json:26]: Isn't this for the maximum value of the position? If so, let's set it to 8 as there are 8 slots available.

File: schema/test_schema.py [hunk 2]
```diff
@@ -58,3 +61,47 @@ def test_schema(self):
...
+        """Tests allocation schema for SOV."""
+        with open("./schema/allocation.schema.json", "r") as f:
+            allocations_schema = json.load(f)
+            allocations: dict[str, Any] = {}
+            allocations.update({"name": "SOV-20230101140000","allocations": []})
+            adm_partner: Partner = Partner.objects.create(
+                name="adm"
+            )
+            kevel_partner: Partner = Partner.objects.create(
+                name="kevel"
```
Comment (tiftran) [inline schema/test_schema.py:74]: humour me, cause we had trouble with the filter schema last time, lets have `k3-v-3l` for this partner name

**Review by tiftran — APPROVED** [review]
just one tiny change

File: schema/test_schema.py (hunk 2 above)
Comment (taddes) [reply schema/test_schema.py:74]: Great call @tiftran !

**Pull request merged**